Source/Core/VideoCommon/Present.cpp

```diff
@@ -329,12 +329,20 @@ float Presenter::CalculateDrawAspectRatio(bool allow_stretch) const
   {
     return SourceAspectRatioToWidescreen(source_aspect_ratio);
   }
-  // For the "custom" mode, we force the exact target aspect ratio, without
-  // acknowleding the difference between the source aspect ratio and 4:3.
   else if (aspect_mode == AspectMode::Custom)
+  {
+    return source_aspect_ratio * (g_ActiveConfig.GetCustomAspectRatio() / (4.0f / 3.0f));
+  }
+  // For the "custom stretch" mode, we force the exact target aspect ratio, without
+  // acknowleding the difference between the source aspect ratio and 4:3.
+  else if (aspect_mode == AspectMode::CustomStretch)
   {
     return g_ActiveConfig.GetCustomAspectRatio();
   }
+  else if (aspect_mode == AspectMode::Raw)
+  {
+    return m_xfb_entry ? 1.f : (static_cast<float>(m_last_xfb_width) / m_last_xfb_height);
```

Comment:
Isn't this the wrong way around?

Reply:
Yes, it was wrong, must have made a mistake when moving things around. Fixed.